From ac172506f857c9813a7a3585441721c2ef8d96a7 Mon Sep 17 00:00:00 2001
From: OpenSUSE riscv <unknown@unknown.unknown>
Date: Thu, 5 Sep 2024 03:17:32 +0200
Subject: [PATCH] LiftoffAssembler f16x8 fixup

---
 .../baseline/riscv/liftoff-assembler-riscv-inl.h | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/wasm/baseline/riscv/liftoff-assembler-riscv-inl.h b/src/wasm/baseline/riscv/liftoff-assembler-riscv-inl.h
index 2970ac1d43c..bac56be6e2a 100644
--- a/src/wasm/baseline/riscv/liftoff-assembler-riscv-inl.h
+++ b/src/wasm/baseline/riscv/liftoff-assembler-riscv-inl.h
@@ -2368,6 +2368,22 @@ void LiftoffAssembler::CallFrameSetupStub(int declared_function_index) {
   CallBuiltin(Builtin::kWasmLiftoffFrameSetup);
 }
 
+bool LiftoffAssembler::emit_f16x8_splat(LiftoffRegister dst,
+                                        LiftoffRegister src) {
+  return false;
+}
+bool LiftoffAssembler::emit_f16x8_extract_lane(LiftoffRegister dst,
+                                               LiftoffRegister lhs,
+                                               uint8_t imm_lane_idx) {
+  return false;
+}
+bool LiftoffAssembler::emit_f16x8_replace_lane(LiftoffRegister dst,
+                                               LiftoffRegister src1,
+                                               LiftoffRegister src2,
+                                               uint8_t imm_lane_idx) {
+  return false;
+}
+
 }  // namespace v8::internal::wasm
 
 #endif  // V8_WASM_BASELINE_RISCV_LIFTOFF_ASSEMBLER_RISCV_INL_H_
-- 
2.39.2

