From 3b6cbb532c403210a925174f904aad3bace01116 Mon Sep 17 00:00:00 2001
From: imported_from_diff <unknown@example.com>
Date: Fri, 7 Jun 2024 08:49:25 +0200
Subject: [PATCH] angle: add riscv64 support

---
 src/common/platform.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/platform.h b/src/common/platform.h
index e31ac1481..b341775ab 100644
--- a/src/common/platform.h
+++ b/src/common/platform.h
@@ -108,7 +108,7 @@
 #endif
 
 // Mips and arm devices need to include stddef for size_t.
-#if defined(__mips__) || defined(__arm__) || defined(__aarch64__)
+#if defined(__mips__) || defined(__arm__) || defined(__aarch64__) || defined(__riscv)
 #    include <stddef.h>
 #endif
 
-- 
2.39.2

