From d96c704f6567758574c3acfaa18625b146b4d04b Mon Sep 17 00:00:00 2001
From: imported_from_diff <unknown@example.com>
Date: Fri, 7 Jun 2024 08:53:25 +0200
Subject: [PATCH] libyuv: riscv changes

---
 BUILD.gn   | 4 ++++
 libyuv.gni | 1 +
 2 files changed, 5 insertions(+)

diff --git a/BUILD.gn b/BUILD.gn
index a72ff065..207cfc36 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -31,6 +31,10 @@ config("libyuv_config") {
   if (!libyuv_use_neon) {
     defines = [ "LIBYUV_DISABLE_NEON" ]
   }
+
+  if (libyuv_disable_rvv) {
+    defines += [ "LIBYUV_DISABLE_RVV" ]
+  }
 }
 
 # This target is built when no specific target is specified on the command line.
diff --git a/libyuv.gni b/libyuv.gni
index 852f08ca..67cd28e4 100644
--- a/libyuv.gni
+++ b/libyuv.gni
@@ -13,6 +13,7 @@ import("//build_overrides/build.gni")
 declare_args() {
   libyuv_include_tests = !build_with_chromium
   libyuv_disable_jpeg = false
+  libyuv_disable_rvv = true
   libyuv_use_neon =
       current_cpu == "arm64" ||
       (current_cpu == "arm" && (arm_use_neon || arm_optionally_use_neon))
-- 
2.39.2

